docs/Tutorials/Tools/Setting-up-VSCode.md

```diff
@@ -0,0 +1,180 @@
+---
+
+What is the most important thing before writing quests?
+Correct! We need a good text editor to make our lives easier with shortcuts and syntax highlighting.
+In this tutorial you will be introduced to Visual Studio Code (VSCode), the most popular text editor.
+
+In this tutorial you will learn how to install and use it to your advantage!
+
+
+<div class="grid" markdown>
+!!! danger "Requirements"
+    * No further requirements.
+
+!!! example "Related Docs"
+    * [YAML](../Getting-Started/Basics/YAML.md)
+</div>
+
+
+## 1. Download and installing VSCode
+
+In this step we will install VSCode on your computer.
+
+_Step 1:_ [Download the right version of Visual Studio Code](https://code.visualstudio.com/download) for your operating system from their website. 
+for your operating system.
+
+---
+_Step 2:_ Click on the VSCodeUserSetup-xXX-X.XX.X.exe in your download folder. (You'll
+find your _downloads_ folder in your file explorer.)
+
+---
+_Step 3:_ Go through the installation steps. The default settings are sufficient for BetonQuest usage.
+If you completed the installation and have the setup is closed. Move on with _Step 4_.
+
+---
+_Step 4:_ Start Visual Studio Code by double-clicking it on your desktop or starting it through your searchbar.
+
+---
+_Step 5:_ In this last step we will activate the `Auto-Save` function from VSCode.
+Go to "File" and then click on "Auto-Save" to activate it.
+
+This is a very useful feature that automatically saves your files. No data will ever be lost again!
+It also saves you from one of the most common errors when writing quests: Forgetting to save!
+From now on, you can just execute `/bq reload` as soon as your mouse leaves the editor.
+
+??? example "VSCode Auto-Save Setup"
+    ![auto_save](../../_media/content/Tutorials/VSCode-Setup/auto_save.png)
+    
+Now that you have a working VS Code installation you can learn how to use it.
+    
+## 2. Creating a workspace in VSCode
+
+If you want to work fast and effectively, it is necessary to create a workspace!
+A workspace is a folder that contains all your project's files. In this case, it will be the BetonQuest folder with all
+quests and settings.
+
+??? tip "More than one workspace"
+    If you want to have more than one plugin in your workspace, you can click on "File" and then "Add Folder to Workspace"
+    to add any other folder.
+    You can also close and reopen the workspace at any time using the steps below.
+    
+
+And this is how to do it:
+
+_Step 1:_ Left-click on the folder button.
+![folder button](../../_media/content/Tutorials/VSCode-Setup/creating_workspace_1.png)
+
+_Step 2:_ Left-click on "Open Folder".
+![Open Folder](../../_media/content/Tutorials/VSCode-Setup/creating_workspace_2.png)
+
+
+_Step 3:_ Define a path and click on "Add".
+
+Navigate to your server files :arrow_right: Go to the "_plugins_" folder :arrow_right: Single-Click the "_BetonQuest_" folder
+:arrow_right: Click "Add" in the bottom right of the dialog.
```

Comment:
this is hard bound to windows. May we want to use a more operating system neutral forulation